src/Aop/Pointcut/PointcutParser.php

```diff
@@ -25,4 +26,12 @@ public function __construct(PointcutGrammar $grammar)
         $parseTable = include 'PointcutParseTable.php';
         parent::__construct($grammar, $parseTable);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function parse(TokenStream $stream)
+    {
+        return new AndPointcut(parent::parse($stream), new ClassDeclaresPointcut());
```

Comment:
You shouldn't touch the `PointcutParser` class at all, it's final :) Instead, all changes should be implemented in grammar and AST transforming, see `PointcutGrammar` class for examples how AST will be converted to the pointcut.